version/version.go

```diff
@@ -0,0 +1,85 @@
+package version
+
+import (
+	"fmt"
+)
+
+var (
+	// Version is the main version number that is being run at the moment.
+	//
+	// Version must be of the format <MAJOR>.<MINOR>.<PATCH>, as described in the semantic versioning specification.
+	Version = "0.4.0"
+
+	// Prerelease is a pre-release marker for the version. If this is "" (empty string) then it means that
+	//it is a final release. Otherwise, this is a pre-release such as "dev" (in development),
+	// "beta", "rc1", etc.
+	Prerelease = "dev"
+
+	// Metadata is any additional (optional) information regarding the build, as described by the semantic
+	// versioning specification.
+	Metadata string
+
+	// GitCommit is the commit associated with the build. It is set by the build process and should not require
+	// human updates under normal circumstances.
+	GitCommit string
+
+	// BuildDate is the date/time when the build was created. It is set by the build process and should not require
+	// human updates under normal circumstances.
+	BuildDate string
+)
+
+// VersionInfo is a container for version information.
+type VersionInfo struct {
```

Comment:
This intermediate struct would allow us to add version details to manifest.json in the future.